sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/CompositeOffset.scala

```diff
@@ -35,19 +37,30 @@ case class CompositeOffset(offsets: Seq[Option[Offset]]) extends Offset {
         case (None, _) => -1
         case (_, None) => 1
        }
-      val signs = comparisons.map(sign).distinct
-      if (signs.size != 1) {
-        throw new IllegalArgumentException(
-          s"Invalid comparison between non-linear histories: $this <=> $other")
+      val nonZeroSigns = comparisons.map(sign).filter { _ != 0}.toSet
+      nonZeroSigns.size match {
+        case 0 => 0                       // if both empty or only 0s
+        case 1 => nonZeroSigns.head       // if there are only (0s and 1s) or (0s and -1s)
+        case _ =>                          // there are both 1s and -1s
+          throw new IllegalArgumentException(
+            s"Invalid comparison between non-linear histories: $this <=> $other")
```

Comment:
@marmbrus This is a bug here. Basically, `[0, 0] < [0, 1]` is correct, but the earlier code would give exception as  `signs = [0, -1]`